client/landing/stepper/declarative-flow/import-hosted-site.ts

```diff
@@ -159,6 +163,7 @@ const importHostedSiteFlow: Flow = {
 						case 'verify-email':
 							return navigate( `verifyEmail?${ urlQueryParams.toString() }` );
 						case 'importer':
+							recordFreeHostingTrialStarted( FlowNames.SiteMigration );
```

Comment:
Recording the reddit event here captures the case where the user has already verified their email, but the migraion flow handles unverified users slightly different from the hosting flow. Unverified migrating users are taken to the `verifyEmail` step, which will be auto-submitted once our polling discovers they're verified now.

One option would be to call `recordFreeHostingTrialStarted()` in the `'verifyEmail'` case below.

Another option would be to move the recording _even closer_ to the click handler. So in `hosting-trial-acknowledge.tsx`, `migration-trial-acknowledge.tsx`, and `import-verify-email/verify-email.tsx`. Maybe that makes it clearer when the event fires, rather than the somewhat ambiguous "when the trialAcknowledge step is submitted"

Or perhaps it makes sense to put it in the network call's success handler after all. Would we want to record the reddit event if they click the button but the network call fails?